connman/src/timezone.c

```diff
@@ -226,18 +226,99 @@ static char *find_origin(void *src_map, struct stat *src_st,
 	return NULL;
 }
 
+static char *get_timezone_alpha2(const char *zone)
+{
+	GIOChannel *channel;
+	struct stat st;
+	char **tokens;
+	char *line;
+	char *alpha2 = NULL;
+	gsize len;
+	int fd;
+
+	if (!zone)
+		return NULL;
+
+	fd = open(USR_SHARE_ZONEINFO_MAP, O_RDONLY | O_CLOEXEC);
+	if (fd < 0) {
+		connman_warn("failed to open zoneinfo map %s",
+							USR_SHARE_ZONEINFO_MAP);
+		return NULL;
+	}
+
+	if (fstat(fd, &st) < 0) {
+		connman_warn("zoneinfo map does not exist");
+		close(fd);
+		return NULL;
+	}
+
+	channel = g_io_channel_unix_new(fd);
+	if (!channel) {
+		connman_warn("failed to create io channel for %s",
+							USR_SHARE_ZONEINFO_MAP);
+		close(fd);
+		return NULL;
+	}
+
+	DBG("read %s for %s", USR_SHARE_ZONEINFO_MAP, zone);
+	g_io_channel_set_encoding(channel, "UTF-8", NULL);
+
+	while (g_io_channel_read_line(channel, &line, &len, NULL, NULL) ==
+							G_IO_STATUS_NORMAL) {
+		if (!line || !*line || *line == '#' || *line == '\n')
+			continue;
```

Comment:
non-null 'line' leaked by continue?

Reply:
Restructured 43d24dac8eeb7b1a50f19f9a4517c84c49cddae8 to address this